packages/frontend/src/hooks/dashboard/useDashboardStorage.tsx

```diff
@@ -1,14 +1,31 @@
-import { type DashboardFilters, type DashboardTile } from '@lightdash/common';
-import { useCallback } from 'react';
+import {
+    type CreateDashboardChartTile,
+    type DashboardFilters,
+    type DashboardTile,
+} from '@lightdash/common';
+import { useCallback, useEffect, useState } from 'react';
 
 const useDashboardStorage = () => {
+    const [isEditingDashboardChart, setIsEditingDashboardChart] =
+        useState(false);
+
     const getIsEditingDashboardChart = useCallback(() => {
         return (
             !!sessionStorage.getItem('fromDashboard') ||
             !!sessionStorage.getItem('dashboardUuid')
         );
     }, []);
 
+    // Update isEditingDashboardChart when storage changes, so that NavBar can update accordingly
+    useEffect(() => {
+        const handleStorage = () => {
+            setIsEditingDashboardChart(getIsEditingDashboardChart());
+        };
+
+        window.addEventListener('storage', handleStorage);
+        return () => window.removeEventListener('storage', handleStorage);
+    }, [getIsEditingDashboardChart]);
```

Comment:
what if we used `useLocalStorage` from mantine in here?

Reply:
I just checked and it basically does the same thing but we don't have to maintain the code

Reply:
I tried to replace all of these with just using Mantine hook, but it didn't work, but I think at the time the issue was more related to how things were done. Will see if I can do this replacement :)

Reply:
I tried to use the `useLocalStorage` and it was messing things up - since this hook uses `sessionStorage` it's a lot to replace... Hope that's ok!